sbi/inference/posteriors/base_posterior.py

```diff
@@ -575,9 +575,21 @@ def __str__(self):
             else ""
         )
 
-        # The net might be sequential because it can have a standardization net. Hence,
-        # we only access its last entry if it is a nn.Sequential.
-        actual_net = self.net[-1] if isinstance(self.net, nn.Sequential) else self.net
+        # For SNRE the net is sequential because it has a standardization net. Hence,
+        # we only access its last entry.
+        try:
+            first_net = self.net[0]
+            # Why not `isinstance(first_net, StandardizeInputs)`? Because
+            # `StandardizeInputs` is defined within a function in
+            # neural_nets/classifier.py and can not be imported here.
+            if first_net.__class__.__name__ == "StandardizeInputs":
+                actual_net = self.net[-1]
```

Comment:
Have you tried this approach instead (it just occured to me):
```
try
   actual_net = self.net[-1]
except TypeError:
   actual_net = self.net
```
wouldn't this cover both the if and the try (as it is now)?

OTOH it might be less clear. At the end of the day, it might be a good idea to encapsulate: `def _get_net_name(self): ...`

Reply:
just using the code above will break things if the net is just a sequential net, but the first entry is simply the first layer.

I moved it to a function in the latest commit

Reply:
Doing this breaks if the net is just a sequential net without standardization layer, e.g. `net=Sequential(nn.Linear, nn.Linear)`.

Moved it to a function in the latest commit.